src/Bicep.Core/TypeSystem/TypeHelper.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Extensions;
+
+namespace Bicep.Core.TypeSystem
+{
+    public static class TypeHelper
+    {
+        /// <summary>
+        /// Try to collapse multiple types into a single (non-union) type. Returns null if this is not possible.
+        /// </summary>
+        public static TypeSymbol? TypeCollapseTypes(IEnumerable<ITypeReference> itemTypes)
```

Comment:
Did you mean `TryCollapseTypes` (guessed from the comment)?

Reply:
Yup 🤦 - fixed